packages/gatsby-transformer-documentationjs/src/gatsby-node.js

```diff
@@ -42,6 +42,64 @@ function prepareDescriptionNode(node, markdownStr, name, helpers) {
   return descriptionNode
 }
 
+exports.sourceNodes = ({ actions }) => {
+  const { createTypes } = actions
+  const typeDefs = `
+    type DocumentationJs implements Node {
+      name: String
+      kind: String
+      memberof: String
+      scope: String
+      access: String
+      readonly: Boolean
+      abstract: Boolean
+      generator: Boolean
+      async: Boolean
+      override: Boolean
+      hideconstructor: Boolean
+      alias: String
+      copyright: String
+      author: String
+      license: String
+      since: String
+      lends: String
+      type: DoctrineType
+      default: JSON
+      augments: [DocumentationJs]
+      examples: [JSON]
```

Comment:
This (and other `JSON` fields in `DoctrineType`) do change schema that will be generated (which is not a blocker - just wanted to mentioned that we would need to bump major as this will be breaking change)

Reply:
I feel that if shape of those types is defineable, it should be fully typed. JSON type should only be used in cases where shape of data is impossible to define.

Reply:
Examples probably can, but I specifically made the ones in DoctrineType JSON because they can be circularly typed, and therefore impossible to query with GraphQL...